### PR TITLE
fix(development) Avoid failing on missing less file to enable dev mode

### PR DESCRIPTION
**Pull request by velcrin**

### CODE REVIEW

File: common/src/test/java/org/bonitasoft/console/common/server/themes/CompilableFileTest.java [hunk 1]
```diff
@@ -40,33 +42,37 @@
...
         final File lessStyle = directory.newFile("style.less");
         FileUtils.writeStringToFile(lessStyle, "@myvar:1px;body{width:@myvar}");
         given(modifier.resolve("style.less")).willReturn(lessStyle);
 
-        final byte[] compilation = new CompilableFile("style.less", "style.css").compile(modifier);
+        compilableFile.compile(modifier);
+        verify(modifier).add(eq("style.css"), captor.capture());
 
-        assertThat(new String(compilation, "UTF-8").replace("\r", "")).isEqualTo("body {\n  width: 1px;\n}\n\n");
+        assertThat(new String(captor.getValue())).isEqualTo("body {\n  width: 1px;\n}\n\n");
```
Comment (redboul) [inline common/src/test/java/org/bonitasoft/console/common/server/themes/CompilableFileTest.java]: does it work on windows?

File: common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java [hunk 2]
```diff
@@ -10,43 +10,52 @@
...
     private final String input;
 
     private final String output;
 
-    public static final CompilableFile[] ALWAYS_COMPILED_FILES = new CompilableFile[] {
+    public static final CompilableFile[] ALWAYS_COMPILED_FILES = new CompilableFile[]{
             new CompilableFile("skin/bootstrap/portal/main.less", "bonita-skin.css"),
     };
 
-    public CompilableFile(final String input, final String output) {
```
Comment (redboul) [inline common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java]: removing public creates an error during web-sp build

File: common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java [hunk 3]
```diff
@@ -10,43 +10,52 @@
...
+    CompilableFile(final String input, final String output) {
         this.input = input;
         this.output = output;
     }
 
-    public byte[] compile(final ThemeArchive.ThemeModifier modifier) {
+    public void compile(final ThemeArchive.ThemeModifier modifier) {
```
Comment (redboul) [inline common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java]: same than above : 
```
bonita-web-sp/server-sp/src/main/java/com/bonitasoft/web/rest/server/datastore/
   theme/PortalThemeInstaller.java:[80,84] error: incompatible types
```

Comment (velcrin) [inline common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java]: This is unfortunate :-)

File: common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java [hunk 4]
```diff
@@ -41,12 +48,15 @@ public CompilableFile(final String input, final String output) {
         try {
             final LessCompiler lessCompiler = new LessCompiler();
             lessCompiler.setEncoding("UTF-8");
-            final byte[] compilation = lessCompiler.compile(modifier.resolve(input)).getBytes("UTF-8");
+            File file = modifier.resolve(input);
+            if (!Files.exists(file.toPath())) {
```
Comment (redboul) [inline common/src/main/java/org/bonitasoft/console/common/server/themes/CompilableFile.java]: Do you want to move this part in a catch block?

**Comment by redboul** [comment]
A Stacktrace is present at dev mode startup:
```
mai 31, 2016 1:49:49 PM org.bonitasoft.console.common.server.themes.CompilableFile compile
Avertissement: Theme compilation failure. File </tmp/bonita_portal_32443@bouboulou/tenants/1/theme/portal/skin/bootstrap/portal/main.less> not found
mai 31, 2016 1:49:49 PM org.bonitasoft.console.server.servlet.PlatformTenantListenerSP initializeDefaultTenant
Grave: Error while initializing the default tenant
org.bonitasoft.engine.exception.BonitaRuntimeException: USERNAME=install | java.lang.NullPointerException
	at org.bonitasoft.engine.api.impl.ServerAPIImpl.wrapThrowable(ServerAPIImpl.java:148)
	at com.bonitasoft.engine.api.impl.ServerAPIExt.wrapThrowable(ServerAPIExt.java:41)
	at org.bonitasoft.engine.api.impl.ServerAPIImpl.invokeMethod(ServerAPIImpl.java:136)
	at org.bonitasoft.engine.api.impl.ClientInterceptor.invoke(ClientInterceptor.java:86)
	at com.sun.proxy.$Proxy60.getLastUpdateDate(Unknown Source)
	at org.bonitasoft.forms.server.ThemeExtractor.getThemeLastUpdateDateFromEngine(ThemeExtractor.java:82)
	at org.bonitasoft.forms.server.ThemeExtractor.retrieveAndExtractCurrentTheme(ThemeExtractor.java:46)
	at org.bonitasoft.console.server.servlet.PlatformTenantListenerSP.initializeDefaultTenant(PlatformTenantListenerSP.java:57)
	at org.bonitasoft.console.common.server.servlet.PlatformTenantListener.contextInitialized(PlatformTenantListener.java:58)
	at org.mortbay.jetty.handler.ContextHandler.startContext(ContextHandler.java:549)
	at org.mortbay.jetty.servlet.Context.startContext(Context.java:136)
	at org.mortbay.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1282)
	at org.mortbay.jetty.handler.ContextHandler.doStart(ContextHandler.java:518)
	at org.mortbay.jetty.webapp.WebAppContext.doStart(WebAppContext.java:499)
	at org.mortbay.jetty.plugin.Jetty6PluginWebAppContext.doStart(Jetty6PluginWebAppContext.java:115)
	at org.mortbay.component.AbstractLifeCycle.start(AbstractLifeCycle.java:50)
	at org.mortbay.jetty.handler.HandlerCollection.doStart(HandlerCollection.java:152)
	at org.mortbay.jetty.handler.ContextHandlerCollection.doStart(ContextHandlerCollection.java:156)
	at org.mortbay.component.AbstractLifeCycle.start(AbstractLifeCycle.java:50)
	at org.mortbay.jetty.handler.HandlerCollection.doStart(HandlerCollection.java:152)
	at org.mortbay.component.AbstractLifeCycle.start(AbstractLifeCycle.java:50)
	at org.mortbay.jetty.handler.HandlerWrapper.doStart(HandlerWrapper.java:130)
	at org.mortbay.jetty.Server.doStart(Server.java:224)
	at org.mortbay.component.AbstractLifeCycle.start(AbstractLifeCycle.java:50)
	at org.mortbay.jetty.plugin.Jetty6PluginServer.start(Jetty6PluginServer.java:132)
	at org.mortbay.jetty.plugin.AbstractJettyMojo.startJetty(AbstractJettyMojo.java:454)
	at org.mortbay.jetty.plugin.AbstractJettyMojo.execute(AbstractJettyMojo.java:396)
	at org.mortbay.jetty.plugin.AbstractJettyRunMojo.execute(AbstractJettyRunMojo.java:210)
	at org.mortbay.jetty.plugin.Jetty6RunMojo.execute(Jetty6RunMojo.java:184)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:106)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:317)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:152)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:555)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:214)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:158)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: java.lang.NullPointerException
	at org.bonitasoft.engine.theme.impl.ThemeServiceImpl.getLastUpdateDate(ThemeServiceImpl.java:216)
	at org.bonitasoft.engine.api.impl.ThemeAPIImpl.getLastUpdateDate(ThemeAPIImpl.java:86)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.bonitasoft.engine.api.impl.ServerAPIImpl.invokeAPI(ServerAPIImpl.java:462)
	at org.bonitasoft.engine.api.impl.ServerAPIImpl$2.call(ServerAPIImpl.java:431)
	at com.bonitasoft.engine.transaction.JTATransactionServiceExt.executeInTransaction(JTATransactionServiceExt.java:49)
	at org.bonitasoft.engine.api.impl.ServerAPIImpl.invokeAPIInTransaction(ServerAPIImpl.java:438)
	at org.bonitasoft.engine.api.impl.ServerAPIImpl.invokeAPI(ServerAPIImpl.java:272)
	at org.bonitasoft.engine.api.impl.ServerAPIImpl.invokeMethod(ServerAPIImpl.java:124)
	... 47 more
```

but it seems to work anyway.

Should we do something about it?

**Pull request merged**